syntax = "proto2"; 

message DropoutLayer {
    optional string name = 1;
    repeated string inputs = 2;
    repeated string outputs = 3;
    enum DropoutMode {
        DROPOUT = 0;
        DROPOUT2D = 1;
        DROPOUT3D = 2;
        ALPHADROPOUT = 3;
    }
    optional DropoutMode layer_mode = 4;
    message dropout_params {
        optional string inplace = 1;
        optional string p = 2;
    }
    optional dropout_params layer_params = 5;
    optional string layer_builder = 6;
}
